lib/rubygems/util/spdx.rb

```diff
@@ -0,0 +1,305 @@
+module Gem
+  class SPDX
+    NONSTANDARD = 'Nonstandard'.freeze
```

Comment:
I like the idea of 'other' and of a way to turn off the warning for a non-spdx license, but I don't really like the word `Nonstandard`.  I read it and think what's the standard it's referring to? How do I know if my license is non-standard?  I'd rather, I think have

```ruby
require "set"
module Gem
  module Licenses
    SPDX = %w[]
    private_constant :SPDX
    COMMERCIAL = ["All rights reserved", "Commercial"]
    private_constant :COMMERCIAL
    OTHER = ["See LICENSE"]
    private_constant :OTHER

    # @api public
    def self.all
      Set.new(SPDX | COMMERCIAL | OTHER)
    end

    # @api public
    def self.non_free
       Set.new(COMMERCIAL | OTHER)
    end

    # @api public
    def self.valid?(license)
      all.include?(license)
    end

    # e.g. given Apache return maybe you mean Apache-2.0
    def self.suggestions_based_on_levenshtein(license)
    end
  end
end
```

and then

```ruby
 def license_warning=(bool)
   @license_warning = bool
 end

 def license_warning
   defined?(@license_warning) && @license_warning || self.license_warning = true
 end

 def licenses= licenses
     @licenses = Array licenses

       raise Gem::InvalidSpecificationException,
           "each license must be 64 characters or less"
       end

      if !Gem::Licenses.valid?(license)
        warning <<-warning
WARNING: license value '#{license}' is invalid.  Use a license identifier from
http://spdx.org/licenses or one of '#{Gem::Licenses.non_free.to_a.join(", ")}'.
You may set `spec.license_warning = false` to turn off this warning.
        warning
      end
     }
 
     warning <<-warning if licenses.empty?
licenses is empty, but is recommended.  Use a license identifier from
http://spdx.org/licenses or one of '#{Gem::Licenses.non_free.to_a.join(", ")}'.
You may set `spec.license_warning = false` to turn off this warning.
     warning
```